common/lib/plugins/aurora_initial_connection_strategy_plugin.ts

```diff
@@ -200,7 +200,9 @@ export class AuroraInitialConnectionStrategyPlugin extends AbstractConnectionPlu
           readerCandidateClient = await connectFunc();
           await this.pluginService.forceRefreshHostList(readerCandidateClient);
           readerCandidate = await this.pluginService.identifyConnection(readerCandidateClient);
-
+          if (isInitialConnection) {
```

Comment:
So we're overriding results of the previous line in case it's an initial connection? I'm not sure I follow the logic.